src/Evolution/DiscontinuousGalerkin/SlopeLimiters/Hierarchical.hpp

```diff
@@ -0,0 +1,1044 @@
+
+  packaged_data->mesh = orientation_map(mesh);
+}
+
+template <size_t VolumeDim, typename... Tags>
+bool Hierarchical<VolumeDim, tmpl::list<Tags...>>::operator()(
+    const gsl::not_null<std::add_pointer_t<db::item_type<Tags>>>... tensors,
+    const Element<VolumeDim>& element, const Mesh<VolumeDim>& mesh,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data) const noexcept {
+  if (UNLIKELY(disable_for_debugging_)) {
+    // Do not modify input tensors
+    return false;
+  }
+  if (UNLIKELY(mesh != Mesh<VolumeDim>(mesh.extents()[0], mesh.basis()[0],
+                                       mesh.quadrature()[0]))) {
+    ERROR("The Hierarchical limiter of type '"
+          << hierarchical_type_
+          << "' does not yet support non-uniform number of collocation points "
+             "in each direction. The extents are: "
+          << mesh.extents());
+  }
+  if (UNLIKELY(alg::any_of(element.neighbors(),
+                           [](const auto& direction_neighbors) noexcept {
+                             return direction_neighbors.second.size() != 1;
+                           }))) {
+    ERROR("The Hierarchical limiter of type '"
+          << hierarchical_type_ << "' does not yet support h-refinement");
+  }
+  alg::for_each(neighbor_data, [&mesh,
+                                this ](const auto& id_packaged_data) noexcept {
+    for (size_t d = 0; d < VolumeDim; ++d) {
+      if (UNLIKELY(id_packaged_data.second.mesh.extents(d) !=
+                   mesh.extents(d))) {
+        ERROR("The Hierarchical limiter of type '"
+              << hierarchical_type_
+              << "' does not yet support p-refinement. Self mesh is: " << mesh
+              << " neighbor mesh is: " << id_packaged_data.second.mesh);
+      }
+      if (UNLIKELY(id_packaged_data.second.mesh.basis(d) != mesh.basis(d))) {
+        ERROR(
+            "Must have the same basis in all dimensions but is not the "
+            "same in dimension: "
+            << d << ". Mesh is: " << mesh
+            << " neighbor mesh is: " << id_packaged_data.second.mesh);
+      }
+      if (UNLIKELY(id_packaged_data.second.mesh.quadrature(d) !=
+                   mesh.quadrature(d))) {
+        ERROR(
+            "Must have the same quadrature in all dimensions but is not "
+            "the same in dimension: "
+            << d << ". Mesh is: " << mesh
+            << " neighbor mesh is: " << id_packaged_data.second.mesh);
+      }
+    }
+  });
+
+  // Compute local modal coefficients
+  Variables<tmpl::list<::Tags::Modal<Tags>...>> coeffs_self(
+      mesh.number_of_grid_points(), 0.0);
+  expand_pack(fill_variables_tag_with_spectral_coeffs<Tags>(&coeffs_self,
+                                                            *tensors, mesh)...);
+
+  bool limited_any_component = false;
```

Comment:
Add comment like `Perform the limiting on the modes` above this line

Reply:
I'm doing `modes` -> `modal coefficients` since I'm sure that change will be requested ;)